### PR TITLE
9327 - backend changes

### PR DESCRIPTION
**Pull request by karthik-aot**
*Issue #:*
https://github.com/bcgov/entity/issues/9327

*Description of changes:*

- Added update task logic for bceid admin
- Added on hold mail template in account-mailer
- Updated create-affidavit logic for reuploading bceid admin affidavit

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-api/src/auth_api/models/task.py [hunk 1]
```diff
@@ -88,3 +88,10 @@ def find_by_task_for_account(cls, org_id, status):
         """Find a task instance that matches the provided id."""
         return db.session.query(Task).filter_by(relationship_id=org_id,
                                                 relationship_type=TaskRelationshipType.ORG.value, status=status).first()
+
+    @classmethod
+    def find_by_task_for_user(cls, org_id, status):
```
Comment (sumesh-aot) [inline auth-api/src/auth_api/models/task.py:93]: name could be find_task_for_user or find_by_user_and_status ?

File: auth-api/src/auth_api/services/affidavit.py [hunk 2]
```diff
@@ -124,6 +128,32 @@ def _modify_task(user):
 
                 remarks = [f'User Uploaded New affidavit .Created New task id: {new_task.identifier}']
                 TaskService.close_task(task_model.id, remarks)
+            # Else, find if there are any associated task of BCEID_ADMIN type, clone and close it
+            else:
+                task_model: TaskModel = TaskModel.find_by_task_for_user(org.id, TaskStatus.HOLD.value)
+                if task_model:
+                    task_type = TaskTypePrefix.BCEID_ADMIN.value
+                    task_info = {'name': org.name,
```
Comment (sumesh-aot) [inline auth-api/src/auth_api/services/affidavit.py:136]: duplicate code can be avoided, as it essentially is cloning task.

File: auth-api/src/auth_api/services/task.py [hunk 3]
```diff
@@ -145,19 +144,40 @@ def _update_relationship(self, origin_url: str = None):
...
-                membership = MembershipModel.find_membership_by_userid(user_id)
-                membership.status = Status.ACTIVE.value
+            if not is_hold:
+                self._update_bceid_admin(is_approved=is_approved, user_id=user_id,
+                                         origin_url=origin_url)
+            else:
                 user: UserModel = UserModel.find_by_id(user_id)
-                user.verified = True
-            # TODO Update affidavit status
+                if user.status != Status.PENDING_STAFF_REVIEW.value:
```
Comment (sumesh-aot) [inline auth-api/src/auth_api/services/task.py:152]: Is this condition necessary? I don't think it's needed.

**Review by sumesh-aot — COMMENTED** [review]
Can you add more test cases plz ?

File: auth-api/src/auth_api/models/task.py (hunk 1 above)
Comment (karthik-aot) [reply auth-api/src/auth_api/models/task.py:93]: I shall refactor as find_by_user_and_status

**Review by sumesh-aot — APPROVED** [review]

**Review by shabeeb-aot — APPROVED** [review]

**Pull request merged**